File: Java/프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = new int[2];
        Map<String, Integer> map = new HashMap<>();
        String target = words[0];
        map.put(target, 1);
        for (int i = 1; i < words.length; i++) {
            if (target.charAt(target.length() - 1) == words[i].charAt(0) && !map.containsKey(words[i])) {
                target = words[i];
            } else {
                answer[0] = (i % n) + 1;
                answer[1] = (i / n) + 1;
                break;
            }
            map.put(words[i], map.getOrDefault(words[i], 0) + 1);
        }

        return answer;
    }
}

File: Python/프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = [0, 0]
    s = set()

    s.add(words[0])
    for i in range(1, len(words)):
        if words[i - 1][-1] != words[i][0] or words[i] in s:
            answer = [(i % n) + 1, (i // n) + 1]
            break
        s.add(words[i])

    return answer